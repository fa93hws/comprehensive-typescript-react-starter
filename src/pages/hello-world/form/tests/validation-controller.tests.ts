import {
  createPasswordValidationController,
  createPasswordConfirmationValidationController,
} from '../validation-controllers';

describe('PasswordValidationController', () => {
  const controller = createPasswordValidationController();
  controller.startValidate();

  it('invalid short password', () => {
    const pwd = new Array(6).fill('1').join('');
    const error = controller.getNormalizedError(pwd);
    expect(error).not.toBeNull();
    expect(error).toMatchSnapshot();
  });

  it('invalid long password', () => {
    const pwd = new Array(10).fill('1').join('');
    const error = controller.getNormalizedError(pwd);
    expect(error).not.toBeNull();
    expect(error).toMatchSnapshot();
  });

  it('valid correct length password', () => {
    const pwd = new Array(8).fill('1').join('');
    const error = controller.getNormalizedError(pwd);
    expect(error).toBeNull();
  });
});

describe('PasswordConfirmationValidationController', () => {
  const getPassword = () => 'password';
  const controller = createPasswordConfirmationValidationController(getPassword);
  controller.startValidate();

  it('invalid short password', () => {
    const pwd = new Array(6).fill('1').join('');
    const error = controller.getNormalizedError(pwd);
    expect(error).not.toBeNull();
    expect(error).toMatchSnapshot();
  });

  it('invalid long password', () => {
    const pwd = new Array(10).fill('1').join('');
    const error = controller.getNormalizedError(pwd);
    expect(error).not.toBeNull();
    expect(error).toMatchSnapshot();
  });

  it('invalid different password', () => {
    const pwd = new Array(8).fill('1').join('');
    const error = controller.getNormalizedError(pwd);
    expect(error).not.toBeNull();
    expect(error).toMatchSnapshot();
  });

  it('valid same password', () => {
    const pwd = 'password';
    const error = controller.getNormalizedError(pwd);
    expect(error).toBeNull();
  });
});
