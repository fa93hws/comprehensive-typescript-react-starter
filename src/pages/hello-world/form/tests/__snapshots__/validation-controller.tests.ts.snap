// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordConfirmationValidationController invalid different password 1`] = `"It should be same as your password"`;

exports[`PasswordConfirmationValidationController invalid long password 1`] = `"Password must not be longer than 10 characters"`;

exports[`PasswordConfirmationValidationController invalid short password 1`] = `"Password must be at least 6 characters long!"`;

exports[`PasswordValidationController invalid long password 1`] = `"Password must not be longer than 10 characters"`;

exports[`PasswordValidationController invalid short password 1`] = `"Password must be at least 6 characters long!"`;
